### PR TITLE
Update to ROCm4.2

### PR DESCRIPTION
**Pull request by jithunnair-amd**

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
@seemethere We should be ready to move ROCm CI to 4.2 now

File: src/main/groovy/ossci/caffe2/DockerVersion.groovy [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // This file is manually managed
 package ossci.caffe2
 class DockerVersion {
-  static final String version = "78bcecef683efc7690aeb31260db64c655b93d56";  // WARNING: if you change this to a new version number, you **MUST** also add that version number to the allDeployedVersions list below
+  static final String version = "cf0725f072e1d52d34c23d848a906fbc4dee44f6";  // WARNING: if you change this to a new version number, you **MUST** also add that version number to the allDeployedVersions list below
```
Comment (jeffdaily) [inline src/main/groovy/ossci/caffe2/DockerVersion.groovy:4]:
```suggestion
  static final String version = "2152ba6309b12251994fcdcd4ff234bfbe646d06";  // WARNING: if you change this to a new version number, you **MUST** also add that version number to the allDeployedVersions list below
```

File: src/main/groovy/ossci/pytorch/DockerVersion.groovy [hunk 2]
```diff
@@ -7,7 +7,7 @@ package ossci.pytorch
 class DockerVersion {
   // WARNING: if you change this to a new version number,
   // you **MUST** also add that version number to the allDeployedVersions list below
-  static final String version = "78bcecef683efc7690aeb31260db64c655b93d56";
+  static final String version = "cf0725f072e1d52d34c23d848a906fbc4dee44f6";
```
Comment (jeffdaily) [inline src/main/groovy/ossci/pytorch/DockerVersion.groovy:10]:
```suggestion
  static final String version = "2152ba6309b12251994fcdcd4ff234bfbe646d06";
```

**Review by jeffdaily — CHANGES_REQUESTED** [review]
`DOCKER_TAG=$(git rev-parse HEAD:.circleci/docker)` has changed since PR was submitted.

2152ba6309b12251994fcdcd4ff234bfbe646d06

**Pull request merged**